File: hrms-frontend/src/services/settlementService.js
// src/services/settlementService.js
import api from '../api/axiosConfig';

const settlementService = {
  getAll: () => api.get('/settlements'),
  getById: (id) => api.get(`/settlements/${id}`),
  getByEmployee: (empId) => api.get(`/settlements/employee/${empId}`),
  getByStatus: (status) => api.get(`/settlements/status/${status}`),
  autoCalculate: (empId) => api.get(`/settlements/auto-calculate/${empId}`),
  create: (data) => api.post('/settlements', data),
  update: (id, data) => api.put(`/settlements/${id}`, data),
  delete: (id) => api.delete(`/settlements/${id}`),

  submit: (id) => api.patch(`/settlements/${id}/submit`),
  approve: (id) => api.patch(`/settlements/${id}/approve`),
  markPaid: (id, data) => api.patch(`/settlements/${id}/mark-paid`, data),
  putOnHold: (id, reason) => api.patch(`/settlements/${id}/hold`, { reason }),

  downloadPdf: (id) => api.get(`/settlements/${id}/pdf`,
    { responseType: 'blob' }),

  getStatistics: () => api.get('/settlements/statistics'),
};

export default settlementService;

File: hrms-backend/src/main/java/com/hrms/service/SettlementService.java

//service/SettlementService.java
package com.hrms.service;

import com.hrms.dto.response.FnFSettlementResponse;
import com.hrms.model.FullFinalSettlement;

import java.math.BigDecimal;
import java.util.List;
import java.util.Map;

public interface SettlementService {

 FnFSettlementResponse create(FullFinalSettlement settlement);
 FnFSettlementResponse update(Long id, FullFinalSettlement settlement);
 FnFSettlementResponse getById(Long id);
 FnFSettlementResponse getByEmployee(Long employeeId);
 List<FnFSettlementResponse> getAll();
 List<FnFSettlementResponse> getByStatus(String status);

 // Auto-calculate
 Map<String, Object> autoCalculate(Long employeeId);

 // Workflow
 FnFSettlementResponse submitForApproval(Long id);
 FnFSettlementResponse approve(Long id, Long approverId);
 FnFSettlementResponse markPaid(Long id, String paymentReference,
                                  String paymentMode);
 FnFSettlementResponse putOnHold(Long id, String reason);
 void delete(Long id);

 // PDF
 byte[] generatePdf(Long id);

 Map<String, Object> getStatistics();
}